### PR TITLE
feat(example): add ITS example and integration tests

### PR DESCRIPTION
**Pull request by Foivos**
[AXE-5184]

[AXE-5184]: https://axelarnetwork.atlassian.net/browse/AXE-5184?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/bcs.ts [hunk 1]
```diff
@@ -193,6 +193,16 @@ function getITSStructs() {
         trusted_addresses: Table,
     });
 
+    const TrustedAddresses = bcs.struct('TrustedAddresses', {
+        trusted_chains: bcs.vector(bcs.string()),
+        trusted_addresses: bcs.vector(bcs.string()),
+    });
+
+    const MessageSetTrustedAddresses = bcs.struct('MessageTrustedAddresses', {
```
[the remark this answers is not in the document]
Comment (npty) [reply src/bcs.ts:201]: this is unused. will remove it

File: test/its.js [hunk 2]
```diff
@@ -0,0 +1,365 @@
...
+                    deployer.toSuiAddress(),
+                ],
+            });
+
+            await txb.signAndExecute(deployer);
+        });
+
+        it('should receive interchain token deployment from other chain successfully', async () => {
+            // Deploy the interchain token and store object ids for treasury cap and metadata
+            const publishReceipt = await publishPackage(client, deployer, 'interchain_token');
```
Comment (milapsheth) [inline test/its.js:308]: allow templating this with the actual name/symbol before deploying?

File: test/its.js [hunk 3]
```diff
@@ -0,0 +1,365 @@
...
+                ],
+            });
+
+            txb.moveCall({
+                target: `${deployments.its.packageId}::service::give_unregistered_coin`,
+                arguments: [objectIds.its, treasuryCap, metadata],
+                typeArguments: [`${publishReceipt.packageId}::q::Q`],
+            });
+
+            await txb.moveCall({
```
Comment (milapsheth) [inline test/its.js:356]: As part of ITS operations task, add a script to use the ITS example via the deployments repo as well

**Review by milapsheth — APPROVED** [review]

File: test/its.js [hunk 4]
```diff
@@ -0,0 +1,365 @@
...
+        });
+
+        const mintReceipt = await tokenTxBuilder.signAndExecute(deployer);
+
+        // Find the coin object id
+        objectIds.coin = findObjectId(mintReceipt, 'its_example::ITS_EXAMPLE');
+
+        await setupGateway();
+
+        await setupGovernance();
```
[the remark this answers is not in the document]
Comment (npty) [reply test/its.js]: do we have to modify [this function](https://github.com/axelarnetwork/axelar-cgp-sui/blob/80f46a0f51005daec61ae55b4512e2ab8e1c1283/move/its/sources/service.move#L352-L356)? cc: foivos

File: test/its.js (hunk 3 above)
Comment (npty) [reply test/its.js]: will do this later

File: move/example/sources/utils.move [hunk 5]
```diff
@@ -0,0 +1,8 @@
+module example::utils;
+
+
```
Comment (milapsheth) [inline move/example/sources/utils.move:3]:
```suggestion
```

Comment (milapsheth) [inline move/example/sources/utils.move:3]: add public function section

File: move/example/sources/utils.move [hunk 6]
```diff
@@ -0,0 +1,8 @@
...
+public fun concat<T: copy>(v1: vector<T>, v2: vector<T>): vector<T> {
+    let mut result = v1;
+    vector::append(&mut result, v2);
```
Comment (milapsheth) [inline move/example/sources/utils.move:6]:
```suggestion
    result.append(v2);
```
does this work?

File: move/example/sources/utils.move [hunk 7]
```diff
@@ -0,0 +1,8 @@
+module example::utils;
+
+
+public fun concat<T: copy>(v1: vector<T>, v2: vector<T>): vector<T> {
```
Comment (milapsheth) [inline move/example/sources/utils.move:4]:
```suggestion
public(package) fun concat<T: copy>(v1: vector<T>, v2: vector<T>): vector<T> {
```

**Review by milapsheth — APPROVED** [review]

File: move/example/sources/its/its.move [hunk 8]
```diff
@@ -39,9 +38,15 @@ public struct ExecutedWithToken has copy, drop {
     amount: u64,
 }
 
-public struct ITS_EXAMPLE has drop {}
+// ------------
+// Capabilities
+// ------------
+public struct ITS has drop {}
```
[the remark this answers is not in the document]
Comment (npty) [reply move/example/sources/its/its.move:44]: this value need to be the same as module name. perhaps change it to something else e.g. its_demo?

**Comment by ecdsafu** [comment]
Hey guys, I can't see an example JS function for sending an interchain transfer in here. Could you point me to an example, or add one in?

**Comment by milapsheth** [comment]
> Hey guys, I can't see an example JS function for sending an interchain transfer in here. Could you point me to an example, or add one in?

Hey, we'll be adding relevant operational scripts in another repo within the next couple of days. We'll keep you posted

File: move/example/sources/its/its.move [hunk 9]
```diff
@@ -41,12 +41,12 @@ public struct ExecutedWithToken has copy, drop {
 // ------------
 // Capabilities
 // ------------
-public struct ITS has drop {}
+public struct ITS_DEMO has drop {}
```
Comment (milapsheth) [inline move/example/sources/its/its.move:44]:
```suggestion
public struct EXAMPLE_COIN has drop {}
```
Or `EXAMPLE_TOKEN`, or `DEMO_COIN`. To make it explicit that it's a coin

**Review by milapsheth — APPROVED** [review]

File: move/example/sources/its/its.move [hunk 10]
```diff
@@ -1,4 +1,4 @@
-module example::its;
+module example::its_demo;
```
[the remark this answers is not in the document]
Comment (npty) [reply move/example/sources/its/its.move:1]: the module and coin struct name are needed to be the same.

File: move/example/sources/its/its.move [hunk 11]
```diff
@@ -16,56 +17,21 @@ use std::ascii::{Self, String};
...
-use sui::coin::{Self, Coin, CoinMetadata, TreasuryCap};
+use sui::coin::Coin;
 use sui::event;
 use sui::hex;
 use sui::sui::SUI;
 
 // -------
 // Structs
 // -------
-public struct Singleton has key {
```
Comment (milapsheth) [inline move/example/sources/its/its.move]: this singleton should still exist here. It's the object associated to this example

File: move/example/sources/its/its.move [hunk 12]
```diff
@@ -16,56 +17,21 @@ use std::ascii::{Self, String};
...
-    id: UID,
-    channel: Channel,
```
Comment (milapsheth) [inline move/example/sources/its/its.move]: The channel is owned by the app, and not the token

File: move/example/sources/its/its.move [hunk 13]
```diff
@@ -107,17 +73,20 @@ public fun register_transaction(
...
 
 /// This function needs to be called first to register the coin for either of
 /// the other two functions to work.
 public fun register_coin(singleton: &Singleton, its: &mut ITS) {
-    let coin_info = coin_info::from_info<ITS_DEMO>(
-        singleton.coin_metadata.get_name(),
-        singleton.coin_metadata.get_symbol(),
-        singleton.coin_metadata.get_decimals(),
-        singleton.coin_metadata.get_decimals(),
+    let coin_metadata = singleton.get_coin_metadata();
```
Comment (milapsheth) [inline move/example/sources/its/its.move:84]: coin meta data should be passed in as input

File: move/example/sources/its/token.move [hunk 14]
```diff
@@ -0,0 +1,66 @@
+module example::token;
+
+use axelar_gateway::channel::{Self, Channel};
+use sui::coin::{Self, CoinMetadata, TreasuryCap};
+
+// -------
+// Structs
+// -------
+public struct Singleton has key {
```
Comment (milapsheth) [inline move/example/sources/its/token.move:9]: there is no need for a singleton here. See the template interchain token package for how this should look like. Coin metadata and treasury cap should be transferred to the deployer. Channel is held by the app's singleton.

File: move/example/sources/its/token.move [hunk 15]
```diff
@@ -0,0 +1,66 @@
...
+public fun get_coin_metadata(self: &Singleton): &CoinMetadata<TOKEN> {
+    &self.coin_metadata
+}
+
+/// -----
+/// Public Functions
+/// -----
+/// Call this to obtain some coins for testing.
+public fun mint(
+    self: &mut Singleton,
```
Comment (milapsheth) [inline move/example/sources/its/token.move:60]: pass in treasury cap to allow minting

**Review by milapsheth — APPROVED** [review]

**Pull request merged**